### ISSUE TITLE
`ec2_instance_type` is not respecting `AWS_MAX_ATTEMPTS` env var

### ISSUE THREAD

**Issue by cbruno10** [turbot/steampipe-plugin-aws#1612]
**Describe the bug**
Queries to the `ec2_instance_type` table do not appear to be respecting the max attempts set through the `AWS_MAX_ATTEMPTS` env variable

**Steampipe version (`steampipe -v`)**
v0.19.0-rc.2

**Plugin version (`steampipe plugin list`)**
v0.93.0

**To reproduce**
Query the `ec2_instance_type` with caching off repeatedly

**Expected behavior**
The API requests should be retried according to the env var

**Additional context**
Related SPC issue: https://github.com/turbotio/steampipe-cloud/issues/1770

**Comment by ParthaI** [comment]
@cbruno10, I could not replicate the issue. I experienced the following observation.

- Set the env variable AWS_MAX_ATTEMPTS with value 2 (`export AWS_MAX_ATTEMPTS=2`)
- Removed `max_error_retry_attempts` config from aws.spc file.
- Ran the steampipe query `select * from aws_ec2_instance_type`

With the above set of configuration the current retry and exponential back-off logic is seems to be working fine with maximum retry attempt `2`.

If we are not setting any max attempt value in either way` environment variable `or `in spc` file, then the default value is 9, but after setting up the AWS_MAX_ATTEMPTS env value to 2 it is retrying the only 2 times. So I concluded that if we are setting up the  AWS_MAX_ATTEMPTS env variable then it is picking up the value from env variable.

I am getting the following error after attempting 2 times.
```
> select * from aws_ec2_instance_type

Error: table 'aws_ec2_instance_type' column 'auto_recovery_supported' requires hydrate data from describeInstanceType, which failed with error operation error EC2: DescribeInstanceTypes, exceeded maximum number of attempts, 2, https response error StatusCode: 503, RequestID: 08f9f38f-6702-4e03-be4c-7dabbbb5c915, api error RequestLimitExceeded: Request limit exceeded..
```
Steampipe plugin version:  aws@latest v0.96.0
Steampipe SDK version: v5.2.0
Steampipe CLI version: v0.19.0-rc.5

I am not sure if I am doing anything wrong here?  Should i try this out with any other set of configuration?
